### DISCUSSION TITLE
Installing past IDAES version

### DISCUSSION THREAD

**Discussion by juanbaquero4** [IDAES/idaes-pse#1455] [Q&A] [answered]
IDAES Team,

Is it possible to install version 2.1.0 of the IDAES platform? If so, could you please provide step-by-step instructions for installing IDAES version 2.1.0?

**Comment by lbianchi-lbl** [comment]
Hi @juanbaquero4, you can try the following steps. Note that this requires `conda`, which you can set up on your system using e.g. [Miniforge](https://github.com/conda-forge/miniforge?tab=readme-ov-file#download)

1. Download the file https://github.com/IDAES/idaes-pse/releases/download/2.1.0/requirements-idaes-pse-2.1.0.txt
2. Open a terminal/command prompt and navigate to the directory where the file was downloaded
3. In the terminal, run:

```sh
conda create --name idaes-2.1.0 python=3.10 --yes
conda activate idaes-2.1.0
pip install -r requirements-idaes-pse-2.1.0.txt
idaes get-extensions
```

**Reply by juanbaquero4** [reply]
thank you @lbianchi-lbl for your answer,

I obtained this Error message during the installation:

(idaes-2.1.0) C:\Users\57313\Desktop\IDAES_2.1>pip install -r requirements-idaes-pse-2.1.0.txt

ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'requirements-idaes-pse-2.1.0.txt'

**Reply by lbianchi-lbl** [reply]
The `requirements-idaes-pse-2.1.0.txt` should be placed in the directory where the command is run. After downloading the `requirements-idaes-pse-2.1.0.txt` file from the provided URL using a browser, you should be able to copy or move it to `C:\Users\57313\Desktop\IDAES_2.1` using e.g. Windows Explorer or any other suitable method.

**Reply by juanbaquero4** [reply]
Thank you again @lbianchi-lbl 

After downloading from GitHub and placing it in the directory where it is running, this Error message appeared:

(idaes-2.1.0) C:\Users\57313\Desktop\IDAES_2.1>pip install -r requirements-idaes-pse-2.1.0.txt

Looking in indexes: https://pypi.python.org/simple/
Obtaining file:///C:/Users/57313/Desktop/IDAES_2.1
ERROR: file:///C:/Users/57313/Desktop/IDAES_2.1 does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

**Reply by lbianchi-lbl** [reply]
@juanbaquero4 I tried running the same commands and could not manage to reproduce your error. Can you double-check that all commands are entered correctly?

**Reply by juanbaquero4** [reply]
hi @lbianchi-lbl, immensely appreciate your answer,

I have done it several times and the same Error appeared; I have uploaded this file in the directory where Idaes is running 
C:\Users\57313\Desktop\IDAES_2.1 :

[requirements-idaes-pse-2.1.0.txt](https://github.com/user-attachments/files/16354886/requirements-idaes-pse-2.1.0.txt) 

These are the commands executed step by step:

```
(base) C:\Users\57313>cd Desktop\IDAES_2.1
(base) C:\Users\57313\Desktop\IDAES_2.1>conda activate idaes-2.1.0
(idaes-2.1.0) C:\Users\57313\Desktop\IDAES_2.1>pip install -r requirements-idaes-pse-2.1.0.txt
```

Looking in indexes: https://pypi.python.org/simple/
Obtaining file:///C:/Users/57313/Desktop/IDAES_2.1
ERROR: file:///C:/Users/57313/Desktop/IDAES_2.1 does not appear to be a Python project: neither 'setup.py' nor 'pyproject.toml' found.

Appreciate your help!

**Reply by lbianchi-lbl** [reply, marked as answer]
@juanbaquero4 the file that you attached to the previous comment is not the correct one. Can you try the following?

1. Delete all copies of the file `requirements-idaes-pse-2.1.0.txt` from your local folders
2. Download the correct file from the following URL: https://github.com/IDAES/idaes-pse/releases/download/2.1.0/requirements-idaes-pse-2.1.0.txt
3. Move or copy the new (correct) local copy of `requirements-idaes-pse-2.1.0.txt` to the `C:\Users\57313\Desktop\IDAES_2.1` folder
4. Open a console window and change directory to `C:\Users\57313\Desktop\IDAES_2.1`
5. Run the `pip install -r requirements-idaes-pse-2.1.0.txt` command again

For a cross-check, you can open `requirements-idaes-pse-2.1.0.txt` using any text editor and verify that its contents are as follows:

```
click==8.1.3
contourpy==1.1.0
cycler==0.11.0
fonttools==4.40.0
idaes-pse==2.1.0
kiwisolver==1.4.4
matplotlib==3.7.1
mpmath==1.3.0
networkx==3.1
numpy==1.25.0
packaging==23.1
pandas==2.0.3
Pillow==9.5.0
Pint==0.22
ply==3.11
Pyomo==6.6.1
pyparsing==3.1.0
python-dateutil==2.8.2
pytz==2023.3
scipy==1.11.1
six==1.16.0
sympy==1.12
typing_extensions==4.7.0
tzdata==2023.3
```

**Reply by juanbaquero4** [reply]
Thank you @lbianchi-lbl !!! now its possible for me to run my old exercises!!!!!!
